### PR TITLE
Recipe to convert explicit getters to the Lombok annotation

### PR DESCRIPTION
**Pull request by timo-a**
## What's changed?

Adds new recipe to convert explicit getter methods to the lombok annotation

## What's your motivation?

- Fix https://github.com/openrewrite/rewrite/issues/3157

## Anything in particular you'd like reviewers to focus on?

Please check especially the naming.

## Anyone you would like to review specifically?
@timtebeek as discussed this is the first of several of [my lombok recipes](https://timo-a.github.io/rewrite-recipe-starter/component-singleton/0/) to be moved to `rewrite-migrate-java`

## Any additional context

This recipe was written before the introduction of traits. I guess the methods of the utils class could be traits nowadays but I'd need really specific hints to use traits here.

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/java/org/openrewrite/java/migrate/lombok/ConvertGetter.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            if (classDeclAfterVisit != classDecl) {
+                //this set collects the fields for which existing methods have already been removed
+                Set<Finding> fieldsToDecorate = getCursor().pollNearestMessage(FIELDS_TO_DECORATE_KEY);
+                doAfterVisit(new FieldAnnotator(fieldsToDecorate));
+            }
+            return classDeclAfterVisit;
+        }
+
+        @Override
+        public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
```
[the remark this answers is not in the document]
Comment (timo-a) [reply src/main/java/org/openrewrite/java/migrate/lombok/ConvertGetter.java:94]: I guess this one broke the build...

**Review by timtebeek — COMMENTED** [review]
Great to see this already @timo-a ! I've added some polish to move the field/method name check to utils as well, and added a failing test for the inner class case not yet covered. Figured you might like the puzzle of solving that one, but do let me know if you'd like some more guidance, or me stepping in for the fix there.

**Comment by timo-a** [comment]
Could you step in, please? I don't have a lot of time at the moment, but I'll try to transfer your fix to the next PRs.

**Review by timtebeek — APPROVED** [review]
Thanks a lot for the work done here @timo-a ! The tests made it easy to simplify the logic. I quite like what we've arrived at now: just a single pass through method declarations, with doAfterVisit to annotate the field based on type, not simple name.

**Comment by timo-a** [comment]
Thanks, I had no idea you could stack `doAfterVisit` calls, this looks much cleaner now!

**Pull request merged**